xradar/src/main/java/com/orzangleli/radar/XRadarView.java

```diff
@@ -251,40 +262,60 @@ public XRadarView(Context context, @Nullable AttributeSet attrs, int defStyleAtt
         percents = new double[count];
         values = new CharSequence[count];
 
+        addDrawTask(this);
+        setLayoutRefreshedListener(new RefreshListener());
+        setTouchEventListener(this);
     }
 
     @Override
-    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
-        radius = Math.min(h, w) / 2 * radarPercent;
-        MAX_TEXT_WIDTH = (int) (Math.min(h, w) / 2 * (1 - radarPercent));
-        //中心坐标
-        centerX = w / 2;
-        centerY = h / 2;
-        postInvalidate();
-        super.onSizeChanged(w, h, oldw, oldh);
-    }
-
-    @Override
-    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
-        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
-        int wSpecSize = MeasureSpec.getSize(widthMeasureSpec);
-        int hSpecSize = MeasureSpec.getSize(heightMeasureSpec);
-
-        setMeasuredDimension(Math.min(wSpecSize, hSpecSize), Math.min(wSpecSize, hSpecSize));
-    }
+    public boolean onTouchEvent(Component component, TouchEvent touchEvent) {
+        HiLog.info(LABEL_LOG, "%{public}s", "onTouchEvent called");
+        switch (touchEvent.getAction()) {
```

Comment:
'switch' statement has too few case labels (1), and should probably be replaced with an 'if' statement

Reply:
Switch statement replaced by if statement.